File: frontend/src/api/cricket.ts
import { fetchApi } from './client'

export interface CricketDetectionResponse {
  document_id: string
  is_scorecard: boolean
  confidence: number
  signals: string[]
  detected_teams: string[]
  summary: string
}

export interface CricketBattingPerformance {
  id: string
  player_name: string
  runs: number
  balls: number
  fours: number
  sixes: number
  strike_rate?: number | null
  dismissal?: string | null
  batting_position?: number | null
  source_page?: number | null
  source_text?: string | null
}

export interface CricketBowlingPerformance {
  id: string
  player_name: string
  overs: number
  maidens: number
  runs_conceded: number
  wickets: number
  economy?: number | null
  wides?: number | null
  no_balls?: number | null
  source_page?: number | null
  source_text?: string | null
}

export interface CricketExtras {
  wides: number
  no_balls: number
  byes: number
  leg_byes: number
  penalty: number
  total: number
}

export interface CricketInnings {
  id: string
  innings_number: number
  team: string
  total_runs: number
  wickets: number
  overs: number
  run_rate?: number | null
  extras: CricketExtras
  batting_performances: CricketBattingPerformance[]
  bowling_performances: CricketBowlingPerformance[]
}

export interface CricketValidation {
  is_valid: boolean
  warnings: string[]
  errors: string[]
}

export interface CricketMatch {
  id: string
  document_id: string
  team_1: string
  team_2: string
  venue?: string | null
  city?: string | null
  match_date?: string | null
  tournament?: string | null
  match_number?: string | null
  format?: string | null
  toss_winner?: string | null
  toss_decision?: string | null
  winner?: string | null
  result_text?: string | null
  player_of_match?: string | null
  innings: CricketInnings[]
  validation?: CricketValidation | null
  created_at: string
  updated_at: string
}

export interface CricketTopPerformer {
  player_name: string
  team: string
  metric_label: string
  metric_value: string
  subtext?: string | null
}

export interface CricketMatchStats {
  document_id: string
  match_id: string
  top_scorers: CricketTopPerformer[]
  top_wicket_takers: CricketTopPerformer[]
  highest_strike_rates: CricketTopPerformer[]
  best_economies: CricketTopPerformer[]
  total_match_runs: number
  total_match_wickets: number
  total_boundaries_fours: number
  total_boundaries_sixes: number
}

export interface CricketMatchSummary {
  document_id: string
  match_id: string
  title: string
  summary_text: string
  highlights: string[]
  winner?: string | null
  player_of_match?: string | null
  generated_at: string
}

export interface CricketPlayerStats {
  player_name: string
  matches_count: number
  innings_batted: number
  total_runs: number
  highest_score: number
  batting_average?: number | null
  batting_strike_rate?: number | null
  fifties: number
  hundreds: number
  total_fours: number
  total_sixes: number
  innings_bowled: number
  total_overs: number
  total_wickets: number
  runs_conceded: number
  bowling_average?: number | null
  bowling_economy?: number | null
  best_bowling_figures?: string | null
}

export async function detectCricketScorecard(documentId: string): Promise<CricketDetectionResponse> {
  return fetchApi<CricketDetectionResponse>(`/api/cricket/documents/${documentId}/detect`, {
    method: 'POST',
  })
}

export async function extractCricketScorecard(documentId: string): Promise<CricketMatch> {
  return fetchApi<CricketMatch>(`/api/cricket/documents/${documentId}/extract`, {
    method: 'POST',
  })
}

export async function getCricketScorecard(documentId: string): Promise<CricketMatch> {
  return fetchApi<CricketMatch>(`/api/cricket/documents/${documentId}`, {
    method: 'GET',
  })
}

export async function getCricketMatchStatistics(documentId: string): Promise<CricketMatchStats> {
  return fetchApi<CricketMatchStats>(`/api/cricket/documents/${documentId}/statistics`, {
    method: 'GET',
  })
}

export async function getCricketMatchSummary(documentId: string): Promise<CricketMatchSummary> {
  return fetchApi<CricketMatchSummary>(`/api/cricket/documents/${documentId}/summary`, {
    method: 'GET',
  })
}

export async function getCricketPlayerStatistics(playerName: string): Promise<CricketPlayerStats> {
  return fetchApi<CricketPlayerStats>(`/api/cricket/players/${encodeURIComponent(playerName)}/statistics`, {
    method: 'GET',
  })
}

File: backend/app/schemas/cricket.py
import uuid
from datetime import datetime
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, ConfigDict

class CricketDetectionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    document_id: uuid.UUID
    is_scorecard: bool
    confidence: float
    signals: List[str]
    detected_teams: List[str]
    summary: str

class CricketBattingPerformanceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    player_name: str
    runs: int
    balls: int
    fours: int
    sixes: int
    strike_rate: Optional[float] = None
    dismissal: Optional[str] = None
    batting_position: Optional[int] = None
    source_page: Optional[int] = None
    source_text: Optional[str] = None

class CricketBowlingPerformanceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    player_name: str
    overs: float
    maidens: int
    runs_conceded: int
    wickets: int
    economy: Optional[float] = None
    wides: Optional[int] = None
    no_balls: Optional[int] = None
    source_page: Optional[int] = None
    source_text: Optional[str] = None

class CricketExtrasResponse(BaseModel):
    wides: int = 0
    no_balls: int = 0
    byes: int = 0
    leg_byes: int = 0
    penalty: int = 0
    total: int = 0

class CricketInningsResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    innings_number: int
    team: str
    total_runs: int
    wickets: int
    overs: float
    run_rate: Optional[float] = None
    extras: CricketExtrasResponse
    batting_performances: List[CricketBattingPerformanceResponse]
    bowling_performances: List[CricketBowlingPerformanceResponse]

class CricketValidationResponse(BaseModel):
    is_valid: bool
    warnings: List[str]
    errors: List[str]

class CricketMatchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    document_id: uuid.UUID
    team_1: str
    team_2: str
    venue: Optional[str] = None
    city: Optional[str] = None
    match_date: Optional[str] = None
    tournament: Optional[str] = None
    match_number: Optional[str] = None
    format: Optional[str] = None
    toss_winner: Optional[str] = None
    toss_decision: Optional[str] = None
    winner: Optional[str] = None
    result_text: Optional[str] = None
    player_of_match: Optional[str] = None
    innings: List[CricketInningsResponse]
    validation: Optional[CricketValidationResponse] = None
    created_at: datetime
    updated_at: datetime

class CricketMatchSummaryResponse(BaseModel):
    document_id: uuid.UUID
    match_id: uuid.UUID
    title: str
    summary_text: str
    highlights: List[str]
    winner: Optional[str] = None
    player_of_match: Optional[str] = None
    generated_at: datetime

class CricketTopPerformer(BaseModel):
    player_name: str
    team: str
    metric_label: str
    metric_value: str
    subtext: Optional[str] = None

class CricketMatchStatsResponse(BaseModel):
    document_id: uuid.UUID
    match_id: uuid.UUID
    top_scorers: List[CricketTopPerformer]
    top_wicket_takers: List[CricketTopPerformer]
    highest_strike_rates: List[CricketTopPerformer]
    best_economies: List[CricketTopPerformer]
    total_match_runs: int
    total_match_wickets: int
    total_boundaries_fours: int
    total_boundaries_sixes: int

class CricketPlayerStatsResponse(BaseModel):
    player_name: str
    matches_count: int
    innings_batted: int
    total_runs: int
    highest_score: int
    batting_average: Optional[float] = None
    batting_strike_rate: Optional[float] = None
    fifties: int
    hundreds: int
    total_fours: int
    total_sixes: int
    innings_bowled: int
    total_overs: float
    total_wickets: int
    runs_conceded: int
    bowling_average: Optional[float] = None
    bowling_economy: Optional[float] = None
    best_bowling_figures: Optional[str] = None
